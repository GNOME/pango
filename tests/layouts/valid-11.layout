{
  "context" : {
    "font" : "serif 12",
    "language" : "en-us",
    "base-gravity" : "south",
    "gravity-hint" : "natural",
    "base-dir" : "weak-ltr",
    "round-glyph-positions" : true,
    "transform" : [
      1,
      0,
      0,
      1,
      0,
      0
    ]
  },
  "text" : "double low error",
  "attributes" : [
    {
      "end" : 6,
      "type" : "underline",
      "value" : "double"
    },
    {
      "end" : 6,
      "type" : "overline",
      "value" : "single"
    },
    {
      "start" : 7,
      "end" : 10,
      "type" : "underline",
      "value" : "low"
    },
    {
      "start" : 7,
      "end" : 10,
      "type" : "strikethrough",
      "value" : true
    },
    {
      "start" : 11,
      "end" : 16,
      "type" : "underline",
      "value" : "error"
    },
    {
      "start" : 11,
      "end" : 16,
      "type" : "rise",
      "value" : 1024
    }
  ],
  "font" : "Cantarell 11",
  "wrap" : "char",
  "width" : 225280,
  "output" : {
    "is-wrapped" : false,
    "is-ellipsized" : false,
    "unknown-glyphs" : 0,
    "width" : 109568,
    "height" : 20480,
    "log-attrs" : [
      {
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "sentence-boundary" : true,
        "sentence-start" : true,
        "backspace-deletes-character" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "line-break" : true,
        "mandatory-break" : true,
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "sentence-boundary" : true,
        "sentence-end" : true,
        "word-boundary" : true
      }
    ],
    "lines" : [
      {
        "start-index" : 0,
        "length" : 16,
        "paragraph-start" : true,
        "direction" : "ltr",
        "runs" : [
          {
            "offset" : 0,
            "length" : 6,
            "text" : "double",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "extra-attributes" : [
              {
                "end" : 6,
                "type" : "underline",
                "value" : "double"
              },
              {
                "end" : 6,
                "type" : "overline",
                "value" : "single"
              }
            ],
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 280,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 0
              },
              {
                "glyph" : 370,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 1
              },
              {
                "glyph" : 438,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 2
              },
              {
                "glyph" : 272,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 3
              },
              {
                "glyph" : 349,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 4
              },
              {
                "glyph" : 287,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 5
              }
            ]
          },
          {
            "offset" : 6,
            "length" : 1,
            "text" : " ",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 1109,
                "width" : 3072,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          },
          {
            "offset" : 7,
            "length" : 3,
            "text" : "low",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "extra-attributes" : [
              {
                "start" : 7,
                "end" : 10,
                "type" : "underline",
                "value" : "low"
              },
              {
                "start" : 7,
                "end" : 10,
                "type" : "strikethrough",
                "value" : true
              }
            ],
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 349,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 0
              },
              {
                "glyph" : 370,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 1
              },
              {
                "glyph" : 467,
                "width" : 11264,
                "is-cluster-start" : true,
                "log-cluster" : 2
              }
            ]
          },
          {
            "offset" : 10,
            "length" : 1,
            "text" : " ",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 1109,
                "width" : 3072,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          },
          {
            "offset" : 11,
            "length" : 5,
            "text" : "error",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "extra-attributes" : [
              {
                "start" : 11,
                "end" : 16,
                "type" : "rise",
                "value" : 1024
              },
              {
                "start" : 11,
                "end" : 16,
                "type" : "underline",
                "value" : "error"
              }
            ],
            "y-offset" : 1024,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 287,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 0
              },
              {
                "glyph" : 409,
                "width" : 6144,
                "is-cluster-start" : true,
                "log-cluster" : 1
              },
              {
                "glyph" : 409,
                "width" : 6144,
                "is-cluster-start" : true,
                "log-cluster" : 2
              },
              {
                "glyph" : 370,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 3
              },
              {
                "glyph" : 409,
                "width" : 6144,
                "is-cluster-start" : true,
                "log-cluster" : 4
              }
            ]
          }
        ]
      }
    ]
  }
}
