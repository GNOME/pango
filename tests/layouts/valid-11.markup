width=220,wrap=char
<span underline='double' overline='single'>double</span> <span underline='low' strikethrough='true'>low</span> <span underline='error' rise='1024'>error</span>
