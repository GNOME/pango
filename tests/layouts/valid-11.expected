double low error

--- parameters

wrapped: 0
ellipsized: 0
lines: 2
width: 225280

--- attributes

range 0 6
[0,6]underline=2
[0,6]overline=1
range 6 7
range 7 10
[7,10]underline=3
[7,10]strikethrough=1
range 10 11
range 11 16
[11,16]underline=4
[11,16]rise=1024
range 16 2147483647

--- directions

0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 

--- cursor positions

0(0) 1(0) 2(0) 3(0) 4(0) 5(0) 6(0) 7(0) 8(0) 9(0) 10(0) 11(0) 12(0) 13(0) 14(0) 15(0) 15(1) 17(0) 

--- lines

i=1, index=0, paragraph-start=1, dir=ltr 'double low error
'
i=2, index=17, paragraph-start=1, dir=ltr ''

--- runs

i=1, index=0, chars=6, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, 'double'
[0,6]underline=2
[0,6]overline=1
i=2, index=6, chars=1, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, ' '
i=3, index=7, chars=3, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, 'low'
[7,10]underline=3
[7,10]strikethrough=1
i=4, index=10, chars=1, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, ' '
i=5, index=11, chars=5, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, 'error'
[11,16]rise=1024
[11,16]underline=4
i=6, index=16, no run, line end
i=7, index=17, no run, line end
